tests/ui/mem_forget.stderr

```diff
@@ -4,19 +4,48 @@ error: usage of `mem::forget` on `Drop` type
 LL |     memstuff::forget(six);
    |     ^^^^^^^^^^^^^^^^^^^^^
    |
+note: argument has type `std::sync::Arc<i32>`
+  --> $DIR/mem_forget.rs:14:22
+   |
+LL |     memstuff::forget(six);
+   |                      ^^^
    = note: `-D clippy::mem-forget` implied by `-D warnings`
 
 error: usage of `mem::forget` on `Drop` type
   --> $DIR/mem_forget.rs:17:5
    |
 LL |     std::mem::forget(seven);
    |     ^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: argument has type `std::rc::Rc<i32>`
+  --> $DIR/mem_forget.rs:17:22
+   |
+LL |     std::mem::forget(seven);
+   |                      ^^^^^
 
 error: usage of `mem::forget` on `Drop` type
   --> $DIR/mem_forget.rs:20:5
    |
 LL |     forgetSomething(eight);
    |     ^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: argument has type `std::vec::Vec<i32>`
+  --> $DIR/mem_forget.rs:20:21
+   |
+LL |     forgetSomething(eight);
+   |                     ^^^^^
+
+error: usage of `mem::forget` on type with `Drop` fields
+  --> $DIR/mem_forget.rs:23:5
+   |
+LL |     std::mem::forget(string);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: argument has type `std::string::String`
+  --> $DIR/mem_forget.rs:23:22
+   |
+LL |     std::mem::forget(string);
+   |                      ^^^^^^
```

Comment:
I find this output, showing the argument, verbose with little benefit. I believe this now shows up because this PR is reusing the lint pass of an existing lint. Do you think there is a simple way to remove the note? Maybe by using `span_lint_and_then` instead?

It's also fine if you disagree, merging it like this should also be fine :)

Reply:
I'm neutral on it, I think seeing "argument has type `std::vec::Vec`" is helpful but not really if it's one of the fields, maybe we can have something like "whose fields implement `Drop`"?

Reply:
Sure, I think the not itself is also fine, the span, showcasing the same statement was the part seems verbose to me. Maybe just make the note unspanned in this case?